RecoHGCal/TICL/plugins/TrackstersPCA.cc

```diff
@@ -56,27 +68,71 @@ void ticl::assignPCAtoTracksters(std::vector<Trackster> &tracksters,
       fillPoint(layerClusters[trackster.vertices(i)], weight);
       for (size_t j = 0; j < 3; ++j)
         barycenter[j] += point[j];
+
+      layerClusterEnergies.push_back(layerClusters[trackster.vertices(i)].energy());
     }
 
     if (energyWeight && trackster.raw_energy())
       barycenter /= trackster.raw_energy();
 
-    // Compute the Covariance Matrix and the sum of the squared weights, used
-    // to compute the correct normalization.
-    // The barycenter has to be known.
-    for (size_t i = 0; i < N; ++i) {
+    LogDebug("TrackstersPCA_Eigen") << "cleaning is  :" << clean << std::endl;
+
+    std::vector<unsigned> filtered_idx;
+    double filtered_energy = 0;
```

Comment:
is it necessary to have double? can we move everything to float? also below

Reply:
I changed to float